internal/pkg/workspace/workspace.go

```diff
@@ -162,25 +165,28 @@ func (ws *Workspace) ListWorkloads() ([]string, error) {
 	})
 }
 
-// Pipeline holds identifying information about a pipeline.
-type Pipeline struct {
-	Name string
-	Path string
+// PipelineManifest holds identifying information about a pipeline manifest file.
+type PipelineManifest struct {
+	Name string // Name of the pipeline inside the manifest file.
+	Path string // Absolute path to the manifest file for the pipeline.
 }
 
 // ListPipelines returns all pipelines in the workspace.
-func (ws *Workspace) ListPipelines() ([]Pipeline, error) {
-	var pipelines []Pipeline
+func (ws *Workspace) ListPipelines() ([]PipelineManifest, error) {
+	var pipelineManifests []PipelineManifest
 	// Look for legacy pipeline.
 	legacyPath, err := ws.PipelineManifestLegacyPath()
 	if err != nil {
 		return nil, err
 	}
 	manifest, err := ws.ReadPipelineManifest(legacyPath)
+	if err != nil && !errors.Is(err, ErrNoPipelineInWorkspace) {
+		return nil, err
+	}
 	if err != nil {
-		log.Infof("Unable to read pipeline manifest at '%s'", legacyPath)
+		ws.logger("Unable to read pipeline manifest at '%s'", legacyPath)
 	}
-	pipelines = append(pipelines, Pipeline{
+	pipelineManifests = append(pipelineManifests, PipelineManifest{
```

Comment:
if `err != nil` then we are still appending the following:
```go
PipelineManifest{Name: "", Path: legacyPath}
```

can we instead change it to:
```go
if err != nil {
   switch {
   case errors.Is(err, ErrNoPipelineInWorkspace):
       ws.logger("Unable to read pipeline manifest: ...")
   default:
       return nil, err
   }
} else {
   pipelineManifests = append(...)
}
```

Reply:
🤦🏼